.buildkite/steps/buildimages.sh

```diff
@@ -1,18 +1,39 @@
 #!/bin/bash
 set -eu
 
-for BUILD_ARCH in amd64 arm32v7 arm64v8; do
+for BUILD_OS in linux darwin; do
+  if [[ $BUILD_OS == "linux" ]]; then
+    for BUILD_ARCH in amd64 arm32v7 arm64v8; do
 cat << EOF
   - label: ":docker: Build Image [${BUILD_ARCH}]"
     command: "authelia-scripts docker build --arch=${BUILD_ARCH}"
     agents:
       build: "true"
     artifact_paths:
       - "authelia-image-${BUILD_ARCH}.tar.zst"
-      - "authelia-linux-${BUILD_ARCH}.tar.gz"
-      - "authelia-linux-${BUILD_ARCH}.tar.gz.sha256"
+      - "authelia-${BUILD_OS}-${BUILD_ARCH}.tar.gz"
+      - "authelia-${BUILD_OS}-${BUILD_ARCH}.tar.gz.sha256"
     env:
       ARCH: "${BUILD_ARCH}"
-    key: "build-docker-${BUILD_ARCH}"
+      OS: "${BUILD_OS}"
+    key: "build-docker-${BUILD_OS}-${BUILD_ARCH}"
 EOF
+    done
+  else
+    for BUILD_ARCH in amd64; do
```

Comment:
I'd rather use a hash table with the OS as the key and the list of architectures as values in order to avoid duplicating the code. It's also more generic since we won't have to change the code if we add new architectures for Darwin or add a new OS.